pocs/camera/camera.py

```diff
@@ -275,23 +280,36 @@ def take_exposure(self,
         if not isinstance(seconds, u.Quantity):
             seconds = seconds * u.second
 
-        self.logger.debug('Taking {} second exposure on {}: {}'.format(
+        self.logger.debug('Taking {} exposure on {}: {}'.format(
             seconds, self.name, filename))
 
-        exposure_event = threading.Event()
         header = self._fits_header(seconds, dark)
 
-        self._take_exposure(seconds=seconds,
-                            filename=filename,
-                            dark=dark,
-                            exposure_event=exposure_event,
-                            header=header,
-                            *args, **kwargs)
+        if not self._exposure_event.is_set():
+            self.logger.warning('Exposure started on {} while one in progress! Waiting.'.format(
+                self))
+            self._exposure_event.wait()
+
+        # Clear event now to prevent any other exposures starting before this one is finsihed.
```

Comment:
The `._exposure_event` is being used several ways. It's being set to signal that an exposure has finished, the data is ready for processing and that the camera is ready to start another exposure. Conversely when it's not set (cleared) that means that the camera is _not_ ready to start another exposure right now.  As it's now a persistent attribute instead of a new `threading.Event` created for each exposure I have to manually `.clear()` it when I starting the set up of a new exposure.

This would all be less confusing once I implement that `.is_ready` property we've discussed.